share/wake/lib/core/examples.wake

```diff
@@ -0,0 +1,191 @@
+            |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "name"
+            | expectOne ".commit.name missing"
+
+        def output = JObject (
+            "message" :-> message,
+            "name" :-> name,
+            Nil
+        )
+
+        Pass output
+
+    single log
+    | Pass
+    |> jFlatten
+    |: findFailFn simpleCommit
+    | rmap JArray
+
+# Simplify full github json log but keep parent urls
+export def example5 (log: JValue): Result JValue Error =
+    def simpleCommit commit =
+        require Pass message =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "message"
+            | expectOne ".commit.message missing"
+
+        require Pass name =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "name"
+            | expectOne ".commit.name missing"
+
+        require Pass parents =
+            single commit
+            | Pass
+            |> jField "parents"
+            |> jFlatten
+            |> jField "html_url"
+            |: makePass JArray
+
+        def output = JObject (
+            "message" :-> message,
+            "name" :-> name,
+            "parents" :-> parents,
+            Nil
+        )
+
+        Pass output
+
+    single log
+    | Pass
+    |> jFlatten
+    |: findFailFn simpleCommit
+    | rmap JArray
+
+# Convert a git log to a simplified wake tuple
+export tuple Commit =
+    Message: String
+    Name: String
+    Parents: List String
+
+# Simplify github json log
+export def example6 (log: JValue): Result (List Commit) Error =
+    def simpleCommit commit =
+        require Pass message =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "message"
+            |> jString
+            | expectOne ".commit.message was missing or not a string"
+
+        require Pass name =
+            single commit
+            | Pass
+            |> jField "commit"
+            |> jField "name"
+            |> jString
+            | expectOne ".commit.name missing or not a string"
+
+        require Pass parents =
+            single commit
+            | Pass
+            |> jField "parents"
+            |> jFlatten
+            |> jField "html_url"
+            |> jString
+
+        def output = Commit message name parents
+
+        Pass output
+
+    # This is a bit complicated IMO but not too bad
+    single log
+    | Pass
+    |> jFlatten
+    |: findFailFn simpleCommit
+
+# This is terrible, no one is going to do this
+export def example7 (json: JValue): Result (List JValue) Error =
+    single json
+    | Pass
+    |> jField "instances"
+    |: (filter (_ | jField "instance_name" |: (jExists "testIndicator".JString | makePass) | getWhenFail False) | makePass)
```

Comment:
This is awful.

Reply:
Couldn't we have a `getIf` analogous to `editIf`? Of type `(JValue => Result Boolean Error) => JValue => Result (List JValue) Error)` ... and then also change `jExists` to `JValue => Result (List JValue) Error => Result Boolean Error`?

=> `|> getIf (_ | jField "instance"name" |> jExists (JString "testIndicator"))`

Reply:
Yeah I think this is sort of the heart of the conflict. We can keep infecting more things with Result to make things work out nicely but it means forgoing pre-existing functions.

But maybe the free error handling is worth it. I think one of the big complaints with traversals was that they introduced additional forms of things (though worse than this because at least these are all still Results).